From: Johannes Berg <johannes@sipsolutions.net>
To: hostap@lists.infradead.org
Subject: trace: handle binutils bfd.h breakage
Date: Wed, 15 Jan 2020 10:10:03 +0100
Message-Id: <1579079403-Ieb75a110fccf593e92e6e0d83bf02ba2e0c86d96@changeid>

Some things in bfd.h that we use were renamed, and in the
case of bfd_get_section_vma() a parameter was dropped.
Work around this.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 src/utils/trace.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/utils/trace.c b/src/utils/trace.c
index 4084343..8f12da8 100644
--- a/src/utils/trace.c
+++ b/src/utils/trace.c
@@ -146,6 +146,17 @@ struct bfd_data {
 	unsigned int line;
 };
 
+/*
+ * binutils removed the bfd parameter and renamed things but
+ * those were macros so we can detect their absence.
+ * Cf. https://sourceware.org/git/gitweb.cgi?p=binutils-gdb.git;a=commitdiff;h=fd3619828e94a24a92cddec42cbc0ab33352eeb4;hp=5dfda3562a69686c43aad4fb0269cc9d5ec010d5
+ */
+#ifndef bfd_get_section_vma
+#define bfd_get_section_vma(bfd, section) bfd_section_vma(section)
+#endif
+#ifndef bfd_get_section_size
+#define bfd_get_section_size bfd_section_size
+#endif
 
 static void find_addr_sect(bfd *abfd, asection *section, void *obj)
 {
